lib/matplotlib/figure.py

```diff
@@ -1144,6 +1146,29 @@ def subplots(self, nrows=1, ncols=1, sharex=False, sharey=False,
             # Returned axis array will be always 2-d, even if nrows=ncols=1.
             return axarr
 
+
+    def subplots_iterator(self, nrows=1, ncols=1, show_in_between=False,
+                          sharex=False, sharey=False, squeeze=True,
+                          subplot_kw=None, gridspec_kw=None):
+        """ Iteratively yields the axis object of a rows x cols subplot and creates new subplots when needed"""
+        while True:
+            axes = self.subplots(
+                nrows=nrows,
+                ncols=ncols,
+                sharex=sharex,
+                sharey=sharey,
+                squeeze=False,
+                subplot_kw=subplot_kw,
+                gridspec_kw=gridspec_kw
+            )
+            assert axes.shape == (nrows, ncols), "Matplotlib panic: the returned shape of subplots() is not what was expected: {0} != {1}".format(axes.shape, (nrows, ncols))
```

Comment:
We removed all asserts from the production code and replaced them with `if cond: raise`.